src/azure-cli/azure/cli/command_modules/backup/custom.py

```diff
@@ -524,10 +526,90 @@ def show_recovery_point(cmd, client, resource_group_name, vault_name, container_
     return client.get(vault_name, resource_group_name, fabric_name, container_uri, item_uri, name)
 
 
+def fetch_tier(paged_recovery_points):
+
+    for rp in paged_recovery_points:
+        isRehydrated = False
+        isInstantRecoverable = False
+        isHardenedRP = False
+        isArchived = False
+
+        if rp.properties.recovery_point_tier_details is None:
+            continue
+
+        for i in range(len(rp.properties.recovery_point_tier_details)):
+            if (rp.properties.recovery_point_tier_details[i].type == "ArchivedRP" and
+                    rp.properties.recovery_point_tier_details[i].status == "Rehydrated"):
+                isRehydrated = True
+
+            if rp.properties.recovery_point_tier_details[i].status == "Valid":
+                if rp.properties.recovery_point_tier_details[i].type == "InstantRP":
+                    isInstantRecoverable = True
+
+                if rp.properties.recovery_point_tier_details[i].type == "HardenedRP":
+                    isHardenedRP = True
+
+                if rp.properties.recovery_point_tier_details[i].type == "ArchivedRP":
+                    isArchived = True
+
+        if (isHardenedRP and isArchived) or (isRehydrated):
+            setattr(rp, "tier_type", "VaultStandardRehydrated")
+
+        elif isInstantRecoverable and isHardenedRP:
+            setattr(rp, "tier_type", "SnapshotAndVaultStandard")
+
+        elif isInstantRecoverable and isArchived:
+            setattr(rp, "tier_type", "SnapshotAndVaultArchive")
+
+        elif isArchived:
+            setattr(rp, "tier_type", "VaultArchive")
+
+        elif isInstantRecoverable:
+            setattr(rp, "tier_type", "Snapshot")
+
+        elif isHardenedRP:
+            setattr(rp, "tier_type", "VaultStandard")
+
+
+def check_rp_move_readiness(paged_recovery_points, target_tier, is_ready_for_move):
+
+    if target_tier and is_ready_for_move is not None:
+        filter_rps = []
+        for rp in paged_recovery_points:
+            if (rp.properties.recovery_point_move_readiness_info is not None and
+                    rp.properties.recovery_point_move_readiness_info['ArchivedRP'].is_ready_for_move ==
+                    is_ready_for_move):
+                filter_rps.append(rp)
+
+        return filter_rps
+
+    if target_tier or is_ready_for_move is not None:
+        raise RequiredArgumentMissingError("""--is-ready-for-move or --target-tier is missing. Please provide
+        the required arguments.""")
+
+    return paged_recovery_points
+
+
+def filter_rp_based_on_tier(recovery_point_list, tier):
```

Comment:
private function naming should start with underscore.

Reply:
done